src/lm_buddy/jobs/_entrypoints/prometheus.py

```diff
@@ -0,0 +1,165 @@
+        ]:
+            raise BadResponseError(f"Score {score} is not in range")
+    except (ValueError, BadResponseError) as e:
+        raise BadResponseError(f"Server returned a malformed response ({e})", e)
+
+    return feedback, score
+
+
+def instruction_to_prompt(instruction: str) -> str:
+    """Given some text containing Prometheus instructions, transform it
+    into an actual llama-2 prompt.
+    """
+    conv = get_conv_template("llama-2")
+    conv.set_system_message("You are a fair evaluator language model.")
+    conv.append_message(conv.roles[0], instruction)
+    conv.append_message(conv.roles[1], None)
+    return conv.get_prompt()
+
+
+def run_eval(
+    config: PrometheusJobConfig, data: Dataset, tokenizer: PreTrainedTokenizer, client: OpenAI
+) -> str:
+    # enable / disable tqdm
+    dataset_iterable = tqdm(data) if config.evaluation.enable_tqdm else data
+
+    # open the output file for writing and iterate on samples
+    tracking_name = config.tracking.name if config.tracking is not None else "output.json"
+    output_fname = Path(config.evaluation.tmp_folder) / tracking_name
+    with output_fname.open("w") as file:
+        for sample in dataset_iterable:
+            # convert instructions from the dataset (`text_field` in a dict) to
+            # prompts that prometheus accepts
+            prompt = instruction_to_prompt(sample[config.dataset.text_field])
+
+            # skip those examples which are too long
+            tokenized_prompt = tokenizer(prompt, truncation=False)
+            if len(tokenized_prompt["input_ids"]) > 3072:
+                continue
+
+            # prepare output
+            result = copy.deepcopy(sample)
+            result["prometheus_output"] = []
+            result["prometheus_score"] = []
+
+            for idx in range(config.evaluation.num_answers):
+                i = 0
+                while i < config.evaluation.max_retries:
+                    try:
+                        response = openai_completion(config, client, prompt)
+                        feedback, score = parse_response(config, response)
+                        break
+                    except (OpenAIError, BadResponseError) as e:
+                        print(f"[w] {e.message}, retrying ({i+1}/{config.evaluation.max_retries})")
+                        i += 1
+                        if i == config.evaluation.max_retries:
+                            raise e
+
+                result["prometheus_output"].append(feedback)
+                result["prometheus_score"].append(score)
+
+            # dump sample results incrementally
+            file.write(json.dumps(result) + "\n")
+
+    # convert plain json dataset in HF format
+    output_hf_name = str(Path(config.evaluation.tmp_folder) / "hf" / tracking_name)
+    ds = load_dataset("json", data_files=str(output_fname), split="train")
+    ds.save_to_disk(output_hf_name)
```

Comment:
question on where these currently get persisted: are they saved to disk on the head node in Ray? If so, are we ok with this file being there rather than in wandb?

Reply:
My rationale is:
- in the ray+wandb scenario, this is persisted locally and then synced to wandb (the file on the pod can then be discarded)
- in the local+wandb scenario, this is persisted locally and then synced to wandb (the local file can then be discarded if desired, which is why the default directory is /tmp)
- in the local+no_tracking scenario, this is just saved locally 
- note that we can have tracking or not on ray and save the datasets to some persisted folders, which is why I allowed users to specify an output folder.

I realised though that what I called `tmp_folder` should be called `output_folder` instead so that it is more clear it can be non-temporary.

We should probably also provide the possibility to choose in the config whether we want to save a reference artifact or the whole file. Ideally it should be part of `tracking` but as it involves all the other entrypoints I think it deserves a separate discussion

Reply:
I think the discussion of persistent disk storage locations is going to be a bigger thing to address in the future. For instance, if we run an entire workflow at once with the new API I am developing, I think it makes sense to have a single folder that houses all the generated artifacts from that run. This will likely involve attributing UIDs of some kind to the workflow as a whole, and to the individual tasks within a workflow.

That being said, I'm not sure what thte best thing to do for now is. We don't really have set storage locations for where this code writes to disk at. The only thing that exists is the `RAY_STORAGE_PATH` env var on our clusters, which exclusively controls the output locations for invocations of Ray train during finetuning.

I think for now, the best we can do is have the user specify on the config an output location. But we will certainly end up revisiting this as the library evolves.

Reply:
Sounds good, let's merge and file a separate issue for this to keep going on the storage in a new ticket